rotkehlchen/db/upgrades/v39_v40.py

```diff
@@ -83,6 +128,114 @@ def _add_base_chain_data(write_cursor: 'DBCursor') -> None:
     log.debug('Exit _add_base_chain_data')
 
 
+def _migrate_ledger_actions(write_cursor: 'DBCursor', conn: 'DBConnection') -> None:
+    """
+    Migrate all ledger actions to history events, so that we can get rid of the
+    deprecated ledger action structure.
+
+    Then remove all no longer needed DB data
+
+    Part of https://github.com/rotki/rotki/issues/6096
+    """
+    log.debug('Enter _migrate_ledger_actions')
+    history_events = []
+    with conn.read_ctx() as cursor:
+        cursor.execute('SELECT timestamp, type, location, amount, asset, rate, rate_asset, link, notes FROM ledger_actions')  # noqa: E501
+        for entry in cursor:
+            location = entry[2]
+            try:
+                val = ord(location)  # can raise TypeError
+                if val < 65 or val > 106:  # not between A and j
+                    raise TypeError
+            except TypeError:
+                log.error(f'Skipped a ledger action from migration to history event due to invalid location {location}')  # noqa: E501
+                continue
+
+            usd_value = ZERO  # try to use a rate asset if given only if it's against USD. Unfortunately this means we can lose data here but there is no way to translate this to a single history event  # noqa: E501
+            if entry[5] is not None and entry[6] is not None and entry[6] == 'USD':
+                try:
+                    amount = deserialize_fval(entry[3], 'amount', 'ledger actions migration')
+                    rate = deserialize_fval(entry[5], 'rate', 'ledger actions migration')
+                except DeserializationError as e:
+                    log.error(f'Skipped a ledger action from migration to history event due {e}')  # noqa: E501
+                    continue
+
+                usd_value = amount * rate
```

Comment:
should we fail here or set price to 0?

Reply:
what do you mean? There is no failure in the 3 lines you quoted

Reply:
is in the `log.error(`. What I mean is that if you fail to deserialize the values for amount and rate instead of doing a continue just the the usd_value = ZERO

Reply:
no if the values of the entry are wrong the entry makes no sense.  How can you have an entry with a non-valid amount?